scripts/py_matter_yamltests/matter_yamltests/definitions.py

```diff
@@ -109,6 +109,18 @@ def get_event_name(self, cluster_id: int, event_id: int) -> str:
         event = self.__get_by_id(cluster_id, event_id, _ItemType.Event)
         return event.name if event else None
 
+    def get_cluster_id_by_name(self, cluster_name: str) -> int:
+        if cluster_name is None:
+            return None
+
+        # The idl parser remove spaces
```

Comment:
Does it remove '/' like in "On/Off" too, or just spaces?

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter_idl/zapxml/handlers/handlers.py#L35

I believe only space:

```py
class ClusterNameHandler(BaseHandler):
    """Handles /configurator/cluster/name elements."""

    def __init__(self, context: Context, cluster: Cluster):
        super().__init__(context, handled=HandledDepth.SINGLE_TAG)
        self._cluster = cluster

    def HandleContent(self, content):
        self._cluster.name = content.replace(' ', '')
```

same for attribute names and such. We may want to start using a common method of 'free form string to variable-compatible name'

Reply:
It does not afaik. It *only* remove spaces. See https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter_idl/zapxml/handlers/handlers.py#L27